### DISCUSSION TITLE
Output styles not generated other then for html files

### DISCUSSION THREAD

**Discussion by franferns** [tailwindlabs/tailwindcss#8823] [Help]
I have a simple set up for adding tailwind css for non html files. The extension am supporting is aspx. The issue is that the output styles are not generated.

I have a folder with two files in a folder src SimpleErrorView.aspx and SimpleErrorView.html below is the set up i run

```
npm init -y
npm install -D tailwindcss
npx tailwindcss init
```

Here's a video of the issue 

https://user-images.githubusercontent.com/779253/178157445-635aaac7-97a4-4533-9533-da6999d1e578.mp4

The output.css file does not generate any styles when the Content is set to SimpleErrorView.aspx . The same set up works when i change the content in the tailwind.config.js to an html file. SimpleErrorView.html.

Is there something that I am missing to get the styles properly generated ?

Thanks & Regards,
Francis P.

**Comment by adamwathan** [comment]
Hey! Need a reproduction here to help.

**Reply by franferns** [reply]
Thanks for your reply . Here is a repo with the issue [error-repo](https://github.com/franferns/tailwind-custom) and i use the tailwind-cli 
to get the output here is the command `npx tailwindcss -i ./src/input.css -o ./dist/output.css --watch` . 

There is also a file with the html extension if i change the extension to html the styles gets generated

**Comment by jjozsi** [comment]
I'm using the latest version of tailwindcss but the issue still persists.
I have a php file with .module extension in the ./modules/DebsTickets/ folder.
Tailwindcss watch doesn't generate styles when I save it.
When I change the file extension to .php, tailwindcss generates styles as expected.
It happens even when I explicitly add {php,module} to the content declarations.

Are the accepted extensions hard coded?
Would it not make more sense to allow any extension and leave it to the users to limit file watching in the content configuration?
This is the content declaration I use:
```
module.exports = {
  content: [
    './templates/*.{html,php}',
    './templates/partials/*.{html,php}',
    './modules/DebsTickets/*.{php,module}',
  ],
```
And out put is generated by this command:
`npx tailwindcss -i './templates/tailwind_theme/custom.css' -o './templates/tailwind_theme/tailwind.css' --jit --watch`

**Reply by franferns** [reply]
@jjozsi  In my case the issue was with the Encoding of the file . The file that I was trying to target with tailwind CSS was using **UTF-16 LE**  changed it to **UTF-8** and it started working for me.

**Reply by jjozsi** [reply]
Thanks, in my case it was already UTF-8.
Watching works simply by changing the extension from .module to .php.
Changing it back to .module stops watching. Consistently.
